test/unit/org/apache/cassandra/cql3/validation/entities/SecondaryIndexTest.java

```diff
@@ -161,6 +170,37 @@ private static String removeQuotes(String indexName)
         return StringUtils.remove(indexName, '\"');
     }
 
+    @Test
+    public void shouldCreateCassandraIndexExplicitly() throws Throwable
+    {
+        createTable("CREATE TABLE %s (userid uuid PRIMARY KEY, firstname text, lastname text, age int)");
+        createIndex("CREATE INDEX byAge ON %s(age) USING 'legacy'");
```

Comment:
The test doesn't verify that the proper index implementation is selected. I mean, the test would pass if a SAI index were created. I think we can do that check with something like:
```java
SecondaryIndexManager indexManager = getCurrentColumnFamilyStore().indexManager;
Index index = indexManager.getIndexByName("byage");
assertTrue(index instanceof CassandraIndex);
```